s3fs/tests/test_s3fs.py

```diff
@@ -1096,6 +1098,27 @@ def test_list_versions_many(s3):
     assert len(versions) == 1200
 
 
+def test_versioned_file_fullpath(s3):
+    versioned_file = versioned_bucket_name + '/versioned_file_fullpath'
+    s3 = S3FileSystem(anon=False, version_aware=True)
```

Comment:
Isn't this per-bucket?

Reply:
This is a per bucket setting, but we are not detecting anywhere right now in s3fs whether a particular bucket has versioning enabled,  consequently it is just mostly used blanketly (similar in the other versioned tests)